File: backend/src/main/java/com/workly/backend/service/ProviderService.java
package com.workly.backend.service;

import com.workly.backend.dto.response.AdminProviderResponse;
import com.workly.backend.dto.response.BookingStatusPointResponse;
import com.workly.backend.dto.response.EarningsPointResponse;
import com.workly.backend.dto.response.ProviderResponse;
import com.workly.backend.dto.response.ProviderReviewResponse;
import com.workly.backend.dto.response.ProviderSummaryResponse;
import com.workly.backend.dto.response.ProviderVerificationResponse;
import com.workly.backend.dto.response.PublicProviderResponse;
import com.workly.backend.dto.response.ScheduleItemResponse;

import java.util.List;

public interface ProviderService {

    ProviderResponse getProviderById(String providerId);

    List<ProviderResponse> getAllProviders();

    ProviderResponse updateProvider(String providerId, ProviderResponse providerResponse);

    void deleteProvider(String providerId);

    List<AdminProviderResponse> getAllProvidersForAdmin();

    AdminProviderResponse getProviderForAdmin(String providerId);

    List<ProviderVerificationResponse> getPendingProviders();

    void approveProvider(String providerId);

    void rejectProvider(String providerId);

    void demoteProvider(String providerId);

    AdminProviderResponse updateProviderSkills(String providerId, List<String> skills);

    /**
     * Verified providers only, for the public browse/booking flow.
     */
    List<PublicProviderResponse> getVerifiedProviders();

    /**
     * A single verified provider's public profile card, looked up by
     * business providerId (e.g. "PRO000001") — used by the /profile/:id page.
     */
    PublicProviderResponse getPublicProviderById(String providerId);

    ProviderResponse getMyProfile();

    ProviderResponse updateMyProfile(String email, ProviderResponse providerResponse);

    /**
     * Dashboard hero + stat cards for the logged-in provider.
     */
    ProviderSummaryResponse getMySummary();

    /**
     * Sum of completed-booking earnings per month, for the last
     * {@code months} months (oldest first), for the logged-in provider.
     */
    List<EarningsPointResponse> getMyEarningsTrend(int months);

    /**
     * Count of the logged-in provider's bookings grouped by status.
     */
    List<BookingStatusPointResponse> getMyBookingStatusBreakdown();

    /**
     * The logged-in provider's accepted, not-yet-completed bookings
     * scheduled within the next {@code days} days.
     */
    List<ScheduleItemResponse> getMySchedule(int days);

    /**
     * The logged-in provider's most recent reviews, newest first,
     * capped at {@code limit}.
     */
    List<ProviderReviewResponse> getMyReviews(int limit);
}

File: frontend/src/services/providerservice.js
import api from "./api";

/**
 * Provider dashboard analytics endpoints (provider-scoped — returns only
 * the logged-in provider's own data, resolved from their auth token).
 */

export const getProviderSummary = () => api.get("/provider/summary");

export const getEarningsTrend = (months = 6) =>
    api.get(`/provider/analytics/earnings?months=${months}`);

export const getProviderBookingStatus = () =>
    api.get("/provider/analytics/booking-status");

export const getProviderReviews = (limit = 5) =>
    api.get(`/provider/reviews?limit=${limit}`);

export const getProviderSchedule = (days = 30) =>
    api.get(`/provider/schedule?days=${days}`);

export const getReviewsForProvider = (id) => api.get(`/providers/${id}/reviews`);

/**
 * Public listing of verified providers, for the browse/booking flow.
 * GET /api/providers
 *   -> [{ providerId, name, service, location, rating, reviews,
 *          jobsDone, verified, skills, avatarUrl }, ...]
 */
export const getVerifiedProviders = () => api.get("/providers");
export const getProviderProfile = (id) => api.get(`/providers/${id}`);

export const getMyProviderProfile = () =>
    api.get("/provider/profile");

export const updateMyProviderProfile = (data) =>
    api.put("/provider/profile", data);